### PR TITLE
Chore/update response handler use

### PR DESCRIPTION
**Pull request by calebkiage**

### CODE REVIEW

File: .azure-pipelines/ci-build.yml [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        inputs:
+          artifactName: Nugets
+          targetPath: $(Build.ArtifactStagingDirectory)
+
+- stage: deploy
+  condition: and(contains(variables['build.sourceBranch'], 'refs/heads/main'), succeeded())
+  dependsOn: build
+  jobs:
+    - deployment: deploy_dotnet_abstractions
+      dependsOn: []
```
Comment (baywet) [inline .azure-pipelines/ci-build.yml:257]:
```suggestion
```

File: .azure-pipelines/ci-build.yml [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        displayName: 'Upload Artifact: Nugets'
+        inputs:
+          artifactName: Nugets
+          targetPath: $(Build.ArtifactStagingDirectory)
+
+- stage: deploy
+  condition: and(contains(variables['build.sourceBranch'], 'refs/heads/main'), succeeded())
+  dependsOn: build
+  jobs:
+    - deployment: deploy_dotnet_abstractions
```
Comment (baywet) [inline .azure-pipelines/ci-build.yml:256]:
```suggestion
    - deployment: deploy_cli_commons
```

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[submodule "kiota-abstractions-dotnet"]
```
Comment (baywet) [inline .gitmodules:1]: why are we relying on a module here and not a package reference?

**Review by baywet — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 3 above)
Comment (calebkiage) [reply .gitmodules:1]: We need the powershell scripts for CI

Comment (baywet) [reply .gitmodules:1]: adding a submodule just for a powerhsell script feels a bit overkill, let's duplicate it?

**Review by baywet — APPROVED** [review]

**Pull request merged**